x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/PersistentCache.java

```diff
@@ -127,6 +136,43 @@ public void removeCacheFile(CacheFile cacheFile) throws IOException {
         getWriter(cacheFile).deleteCacheFile(cacheFile);
     }
 
+    public long getCacheSize(ShardId shardId, SnapshotId snapshotId) {
+        long aggregateSize = 0L;
+        final CacheIndexWriter writer = writers.get(0);
```

Comment:
There is a persistent cache Lucene index per node data path and a data node can have more than one data paths. A shard is assigned to one of the data path and its cache files will be indexed into the persistent cache Lucene index that exist on that node path. 

What you can do here is iterating over CacheIndexWriter writers and stop if the search query returns at least a document.

Reply:
:facepalm: right that makes more sense, thanks for spotting!